experimental/devices/mfrc522/commands/low_level.go

```diff
@@ -0,0 +1,293 @@
+			break
+		}
+	}
+	lsb, err := r.DevRead(CRCResultRegL)
+	if err != nil {
+		return nil, err
+	}
+
+	msb, err := r.DevRead(CRCResultRegM)
+	if err != nil {
+		return nil, err
+	}
+	return []byte{lsb, msb}, nil
+}
+
+// SetBitmask sets register bit.
+func (r *LowLevel) SetBitmask(address, mask int) error {
+	current, err := r.DevRead(address)
+	if err != nil {
+		return err
+	}
+	return r.DevWrite(address, current|byte(mask))
+}
+
+// ClearBitmask clears register bit.
+func (r *LowLevel) ClearBitmask(address, mask int) error {
+	current, err := r.DevRead(address)
+	if err != nil {
+		return err
+	}
+	return r.DevWrite(address, current&^byte(mask))
+}
+
+// StopCrypto stops the crypto chip.
+func (r *LowLevel) StopCrypto() error {
+	return r.ClearBitmask(Status2Reg, 0x08)
+}
+
+// WaitForEdge waits for an IRQ pin to strobe.
+func (r *LowLevel) WaitForEdge(timeout time.Duration) bool {
+	return r.irqPin.WaitForEdge(timeout)
+}
+
+// Auth authenticate the card fof the sector/block using the provided data.
+//
+// 	mode - the authentication mode.
+// 	sector - the sector to authenticate on.
+// 	block - the block within sector to authenticate.
+// 	sectorKey - the key to be used for accessing the sector data.
+// 	serial - the serial of the card.
+func (r *LowLevel) Auth(mode byte, blockAddress byte, sectorKey [6]byte, serial []byte) (AuthStatus, error) {
+	buffer := make([]byte, 2)
+	buffer[0] = mode
+	buffer[1] = blockAddress
+	buffer = append(buffer, sectorKey[:]...)
+	buffer = append(buffer, serial[:4]...)
+	_, _, err := r.CardWrite(PCD_AUTHENT, buffer)
+	if err != nil {
+		return AuthReadFailure, err
+	}
+	if n, err := r.DevRead(Status2Reg); err != nil || n&0x08 == 0 {
+		return AuthFailure, err
+	}
+	return AuthOk, nil
+}
+
+// CardWrite the low-level interface to write some raw commands to the card.
+//
+// 	command - the command register
+// 	data - the data to write out to the card using the authenticated sector.
+func (r *LowLevel) CardWrite(command byte, data []byte) ([]byte, int, error) {
+	var backData []byte
+	backLength := -1
+	irqEn := byte(0x00)
+	irqWait := byte(0x00)
+
+	switch command {
+	case PCD_AUTHENT:
+		irqEn = 0x12
+		irqWait = 0x10
+	case PCD_TRANSCEIVE:
+		irqEn = 0x77
+		irqWait = 0x30
+	}
+
+	if err := r.DevWrite(CommIEnReg, irqEn|0x80); err != nil {
+		return nil, -1, err
+	}
+	if err := r.ClearBitmask(CommIrqReg, 0x80); err != nil {
+		return nil, -1, err
+	}
+	if err := r.SetBitmask(FIFOLevelReg, 0x80); err != nil {
+		return nil, -1, err
+	}
+	if err := r.DevWrite(CommandReg, PCD_IDLE); err != nil {
+		return nil, -1, err
+	}
+
+	for _, v := range data {
+		if err := r.DevWrite(FIFODataReg, v); err != nil {
+			return nil, -1, err
+		}
+	}
+
+	if err := r.DevWrite(CommandReg, command); err != nil {
+		return nil, -1, err
+	}
+
+	if command == PCD_TRANSCEIVE {
+		if err := r.SetBitmask(BitFramingReg, 0x80); err != nil {
+			return nil, -1, err
+		}
+	}
+
+	i := 2000
+	n := byte(0)
+
+	for ; i > 0; i-- {
```

Comment:
What about a time based check? This code is literally dependent on the I²C bus speed...